addon/test-waiter.ts

```diff
@@ -0,0 +1,116 @@
+import { IWaiter, WaiterName, ITestWaiterDebugInfo } from './types';
+import { register } from './waiter-manager';
+
+/**
+ * A class providing creation, registration and async waiting functionality.
+ *
+ * @public
+ * @class SimpleWaiter<T>
+ *
+ * @example
+ *
+ * import Component from '@ember/component';
+ * import { SimpleWaiter } from 'ember-test-waiters';
+ *
+ * if (DEBUG) {
+ *   let waiter = new SimpleWaiter('friend-waiter');
+ * }
+ *
+ * export default class Friendz extends Component {
+ *   didInsertElement() {
+ *     waiter.beginAsync(this);
```

Comment:
Usages of `waiter` should also be guarded by `DEBUG`:

```suggestion
       DEBUG && waiter.beginAsync(this);
```